File: 1233 Remove Sub-Folders from the Filesystem/1233.cpp
class Solution {
public:
    vector<string> removeSubfolders(vector<string>& folder) {
        sort(folder.begin(), folder.end());       
        vector<string> ans;

        for (string f: folder){
            if (ans.empty() || !f.starts_with(ans.back() + "/")) {
                ans.push_back(f);
            }
        }

        return ans;
    }
};

File: 1233 Remove Sub-Folders from the Filesystem/1233.py
class Solution:
    def removeSubfolders(self, folder: List[str]) -> List[str]:
        folder.sort()
        ans = []
        prev = " "

        for f in folder:
            if not f.startswith(prev + '/'):
                ans.append(f)
                prev = f

        return ans
